core/server/src/main/java/alluxio/worker/netty/DataServerReadHandler.java

```diff
@@ -176,39 +177,37 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
   }
 
   /**
-   * Requires to hold mLock when calling this.
-   *
    * @return true if there are too many packets in-flight
    */
+  @GuardedBy("mLock")
   private boolean tooManyPendingPackets() {
     return mPosToQueue - mPosToWrite >= MAX_PACKETS_IN_FLIGHT * PACKET_SIZE;
   }
 
   /**
-   * Requires to hold mLock when calling this.
-   *
    * @return true if we should restart the packet reader
    */
+  @GuardedBy("mLock")
   private boolean shouldRestartPacketReader() {
     return !mPacketReaderActive && !tooManyPendingPackets() && mPosToQueue < mRequest.end();
   }
 
   /**
-   * Requires to hold mLock when calling this.
-   *
    * @return the number of bytes remaining to push to the netty queue. Return 0 if it is cancelled
    */
+  @GuardedBy("mLock")
   private long remainingToQueue() {
-    return mRequest.end() - mPosToQueue;
+    ReadRequestInternal request = mRequest;
+    return (request == null || mPosToQueue == -1) ? 0 : request.end() - mPosToQueue;
```

Comment:
mPosToQueue == -1 is not strictly necessary as mRequest is reset before mPosToQueue. But just to be safer